### PR TITLE
Add builtin 'config' object and 'features.future_geometry' toggle

### PR DESCRIPTION
**Pull request by djhoese**
This is the first step of many to allow users a way of transitioning easily between pyresample 1.x and pyresample 2.x interfaces and objects. As a first step this PR adds a new `pyresample.config` object using `donfig` and includes a `features.future_geometries` boolean flag. This flag modifies utility functions like `create_area_def` to return a future `AreaDefinition` instead of the old/existing `AreaDefinition`. Along with some changes to imports users should be able to easily switch usage to the new geometry objects semi-easily (I think).

This also adds a `to_legacy()` method to the future area so it is easier to create the future area but return the legacy version.

As part of these changes I originally added an import in `pyresample/__init__.py` so that importing `AreaDefinition` and `SwathDefinition` from `pyresample` would get you the new versions of the classes if this toggle config flag was set. This seemed near impossible to test so I ended up removing it.

TODO:
 - [x] Add documentation on configuration
 - [x] Modify other utility methods that create AreaDefinitions to use the toggle

 - [ ] Closes #xxxx 
 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``git diff origin/main **/*py | flake8 --diff`` 
 - [x] Fully documented

### CODE REVIEW

File: docs/source/howtos/configuration.rst [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+1. Environment variable
+2. YAML file
+3. At runtime with ``pyresample.config``
+
+This functionality is provided by the :doc:`donfig <donfig:configuration>`
+library. The currently available settings are described below.
+Each option is available from all three methods. If specified as an
+environment variable or specified in the YAML file on disk, it must be set
+**before** Satpy is imported.
```
Comment (mraspaud) [inline docs/source/howtos/configuration.rst:18]: Should it be `pyresample` instead of `Satpy`?

File: pyresample/test/conftest.py [hunk 2]
```diff
@@ -38,6 +38,18 @@
 DST_AREA_SHAPE = (80, 85)
 
 
+@pytest.fixture(autouse=True)
+def reset_satpy_config(tmpdir):
```
Comment (mraspaud) [inline pyresample/test/conftest.py:42]: pyresample?

**Review by mraspaud — COMMENTED** [review]
A couple of typos I think, but otherwise LGTM! Thanks for taking the time to cleanup on the way

File: pyresample/test/conftest.py (hunk 2 above)
Comment (djhoese) [reply pyresample/test/conftest.py]: fixed

**Pull request merged**